app/models/post.rb

```diff
@@ -148,17 +148,18 @@ def nsfw
     self.author.profile.nsfw?
   end
 
-  def self.find_by_guid_or_id_with_user(id, user=nil)
-    key = id.to_s.length <= 8 ? :id : :guid
-    post = if user
-             user.find_visible_shareable_by_id(Post, id, :key => key)
-           else
-             Post.where(key => id).includes(:author, :comments => :author).first
-           end
-
-    # is that a private post?
-    raise(Diaspora::NonPublic) unless user || post.try(:public?)
+  def self.find_public(id)
+    post = Post.where(Post.key_sym(id) => id).includes(:author, comments: :author).first
+    post.try(:public?) || raise(Diaspora::NonPublic)
```

Comment:
Perfect example of why I dislike that style, `raise Diaspora::NonPublic unless post.try?(:public?)` is a lot more readable IMO.